com.unity.netcode.gameobjects/Editor/CodeGen/NetworkBehaviourILPP.cs

```diff
@@ -595,8 +595,8 @@ private void ProcessNetworkBehaviour(TypeDefinition typeDefinition, string[] ass
 
             // override NetworkBehaviour.__getTypeName() method to return concrete type
             {
-                var baseType = typeDefinition.BaseType.Resolve();
-                var baseGetTypeNameMethod = baseType.Methods.First(p => p.Name.Equals(nameof(NetworkBehaviour.__getTypeName)));
+                var networkBehaviourType = m_NetworkBehaviour_TypeRef.Resolve();
+                var baseGetTypeNameMethod = networkBehaviourType.Methods.First(p => p.Name.Equals(nameof(NetworkBehaviour.__getTypeName)));
```

Comment:
there's a slight difference between typeref and typedef, in this case we're trying to resolve typedef from typedef. it'd be good to name this as "-TypeDef" to make it super obvious that we're dealing with typedef not typeref here.
```suggestion
                var networkBehaviourTypeDef = m_NetworkBehaviour_TypeRef.Resolve();
                var baseGetTypeNameMethod = networkBehaviourTypeDef.Methods.First(p => p.Name.Equals(nameof(NetworkBehaviour.__getTypeName)));
```